CMakeLists.txt

```diff
@@ -966,6 +967,18 @@ target_sources(
     ${INCLUDE_FILES}
     ${SRC_FILES}
 )
+
+if (MBGL_WITH_CLANG_TIDY)
+    find_program(CLANG_TIDY_COMMAND NAMES clang-tidy)
+    if(NOT CLANG_TIDY_COMMAND)
+        message(FATAL_ERROR "ENABLE_CLANG_TIDY is ON but clang-tidy is not found!")
+    else()
+        message(STATUS "Found clang-tidy at ${CLANG_TIDY_COMMAND}")
+    endif()
+
+    set_target_properties(mbgl-core PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND};--extra-arg=-Wno-error=unknown-warning-option")
```

Comment:
Which warning option is uknown?

Reply:
`error: unknown warning option '-Werror=maybe-uninitialized'; did you mean '-Werror=uninitialized'?`

I guess this option is only available in GCC. There is a way to check that flag is supported by compiler before adding it in CMake(i.e. we could fix that), but let's do things step by step please :) 

https://github.com/Project-OSRM/osrm-backend/blob/0ca913132acea46e0c56e4665af50d68f53eec68/cmake/warnings.cmake#L5

Reply:
Sure, done.